articles/kotlin-ranges/index.md

```diff
@@ -0,0 +1,80 @@
+### Kotlin Range Operators
+Kotlin provides a wide range of solutions. Most of these solutions aim to make development easier and faster. It provides easier ways to perform specific actions. Since it's compiled the same way as Java, its functions and methods ensure your code is more efficient. The boilerplate code present in Java is also greatly reduced. This article addresses one of the solutions it offers in iterations. It will go through Kotlin ranges.
+
+### What are ranges
+A range is a collection of values determined by a start and stop value. A simple example is, values between 1 and 5 form a range. That is, a range of 1 as the start value and 5 as the end value. Both the start and end values belong to the range too. In other words, a range is an interval from the start value to end value. We can use functions to create progressions and check for values in ranges. Let's look at how we could use ranges.  
+
+Before we get to write code, you can use the following repl to run the functions. All you need to do is call the specific function in the `main` function. Then run the code and your output will be printed 🤓.
+
+<iframe height="400px" width="100%" src="https://repl.it/@Linusmuema/Ranges?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+### 1. To create a progression
+We can use the range operator to create an iterable range. A loop is created that goes through each item from the start value to the end value. This helps us to be sure of the number of iterations in the loop. It gives the developer more control over the loop. In the repl, call the `printNumRange` function and run. The output should be
+
+```bash
+1,2,3,4,5,
+```
+
+This loop goes from 1 to 5. But that is not the only trick ranges have. We can also create ranges using characters. We can create a range that starts from a specific letter in the alphabet to another. This can be from `a` to `e`. You can go ahead and call the `printCharRange` function. The expected output is
+
+```bash
+a,b,c,d,e,
+```
+It was not easy to do this in java. When implemented it led to lots of boilerplate and dirty code.
```

Comment:
You can give the equivalent of this in Java and explain why it is dirty code and contrast how there is lesser code in Kotlin